rad/src/reducers/hash_concatenate.rs

```diff
@@ -0,0 +1,197 @@
+use crate::{
+    error::RadError,
+    hash_functions::RadonHashFunctions,
+    operators::bytes as bytes_operators,
+    reducers::RadonReducers,
+    types::{array::RadonArray, bytes::RadonBytes, RadonType, RadonTypes},
+};
+use serde_cbor::value::Value;
+
+/// Reducer that concatenate all the RadonTypes and hash
+pub fn hash_concatenate(input: &RadonArray) -> Result<RadonTypes, RadError> {
+    let value = input.value();
+
+    match value.first() {
+        None => Err(RadError::UnsupportedReducer {
+            array: input.clone(),
+            reducer: RadonReducers::HashConcatenate.to_string(),
+        }),
+        Some(RadonTypes::Bytes(_)) => {
+            let concatenated_bytes = value.iter().try_fold(
+                Vec::with_capacity(32 * value.len()),
+                |mut bytes, item| match item {
+                    RadonTypes::Bytes(rad_bytes) => {
+                        let new_bytes = rad_bytes.value();
+
+                        // Take up to 32 bytes from new_bytes
+                        let new_bytes_len = std::cmp::min(new_bytes.len(), 32);
+                        bytes.extend_from_slice(&new_bytes[..new_bytes_len]);
+                        // In case of less than 32 bytes, it will be zero-padded
+                        for _i in new_bytes_len..32 {
+                            bytes.push(0);
+                        }
```

Comment:
This is LSB padding, which changes the Big-Endian value of the bytes. I'd prefer MSB padding (i.e. pushing the zeroes first).

Reply:
This is fixed, now extend is called after the zeros.